packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/BillingManager.java

```diff
@@ -253,27 +277,57 @@ public int getBillingClientResponseCode() {
   }
 
   /**
-   * Query purchases across various use cases and deliver the result in a formalized way through
-   * a listener
+   * Query both in app purchases and subscriptions and deliver the result in a formalized way 
+   * through a listener
    */
   public void queryPurchases(final Promise promise) {
     Runnable queryToExecute = new Runnable() {
+      @NonNull
+      private BillingResult aggregateBillingResults(@NonNull Set<BillingResult> billingResults) {
+        for (BillingResult result: billingResults) {
+          if (result.getResponseCode() != BillingResponseCode.OK) {
+            return result;
+          }
+        }
+        return billingResults.iterator().next();
+      }
```

Comment:
Move this out to be a private method of the BillingManager since the method doesn't need to close over any variables. When we convert this code to Kotlin it will make the Runnable nicer to write (Kotlin has special syntactic sugar for classes with only one method so they can be written more like JS callbacks).